vtex-apps/store-newsletter
# Add support for custom newsletter fields

#### What problem is this solving?

This enables users to add custom fields to their client's newsletter subscription info, such as the binding associated with the subscription, through the use of a new block called `newsletter-hidden-field`.

#### How to test it?

This branch is linked at [Workspace](https://victormiranda--storecomponents.myvtex.com/). Try subscribing to the newsletter, at the bottom of the page. Then check the mutation variables that we sent with the `subscribeNewsletter` mutation using Apollo Dev Tools.

#### Screenshots or example usage:

Using the following configuration

```json
{
  "newsletter-form": {
    "children": [
      "rich-text#newsletter-title",
      "flex-layout.row#newsletter-inputs",
      "newsletter-hidden-field"
    ],
    "props": {
      "SuccessState": "flex-layout.row#newsletter-success"
    }
  },

  "flex-layout.row#newsletter-inputs": {
    "children": [
      "newsletter-input-email",
      "newsletter-input-name",
      "newsletter-submit"
    ],
    "props": {
      "horizontalAlign": "center",
      "preventHorizontalStretch": true
    }
  },

  "newsletter-hidden-field": {
    "props": {
      "dynamicFields": [
         "bindingId",
         "bindingUrl"
      ]
    }
  },
}
```

this is what gets sent:

![Screen Shot 2021-01-12 at 16 23 47](https://user-images.githubusercontent.com/27777263/104362475-9eebbb80-54f2-11eb-9439-f27e2552ae9f.png)


#### Related to / Depends on

Depends on https://github.com/vtex-apps/store-graphql/pull/520 and https://github.com/vtex-apps/native-types/pull/32.

#### Notes

This PR was developed with Arian's help! 🎉  @ariangc

#### How does this PR make you feel? [:link:](http://giphy.com/)



![](https://media.giphy.com/media/0Av9l0VIc01y1isrDw/giphy.gif)
